x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/connector_detail/attach_index_box.tsx

```diff
@@ -65,66 +65,109 @@ export const AttachIndexBox: React.FC<AttachIndexBoxProps> = ({ connector }) =>
     createApiError,
     attachApiError,
   } = useValues(AttachIndexLogic);
+
+  const { makeRequest } = useActions(FetchAvailableIndicesAPILogic);
+  const { data, status } = useValues(FetchAvailableIndicesAPILogic);
+  const isLoading = [Status.IDLE, Status.LOADING].includes(status);
+
+  // Helper function to remove the managed connector index prefix from the index name
+  const removePrefixConnectorIndex = (indexName: string) => {
+    if (!connector.is_native) {
+      return indexName;
+    }
+    if (indexName.startsWith(MANAGED_CONNECTOR_INDEX_PREFIX)) {
+      return indexName.substring(MANAGED_CONNECTOR_INDEX_PREFIX.length);
+    }
+    return indexName;
+  };
+
+  // Helper function to add the managed connector index prefix to the index name
+  const prefixConnectorIndex = (indexName: string) => {
+    if (!connector.is_native) {
+      return indexName;
+    }
+    if (indexName.startsWith(MANAGED_CONNECTOR_INDEX_PREFIX)) {
+      return indexName;
+    }
+    return `${MANAGED_CONNECTOR_INDEX_PREFIX}${indexName}`;
+  };
+
+  const [query, setQuery] = useState<{
+    isFullMatch: boolean;
+    searchValue: string;
+  }>();
+  const [sanitizedName, setSanitizedName] = useState<string>(
+    prefixConnectorIndex(formatApiName(connector.name))
+  );
+
   const [selectedIndex, setSelectedIndex] = useState<
     { label: string; shouldCreate?: boolean } | undefined
   >(
     connector.index_name
       ? {
-          label: connector.index_name,
+          label: removePrefixConnectorIndex(connector.index_name),
```

Comment:
We want to render current selection without `content-` prefix, as this is fixed part in UI